### DISCUSSION TITLE
How do I install the ml_boards.h library?

### DISCUSSION THREAD

**Discussion by mguibord** [marcel-licence/esp32_midi_sampler#56] [Q&A] [answered]
How do I install the ml_boards.h library?

**Comment by marcel-licence** [comment, marked as answer]
Hi, the header is part of the ML_SynthTools -> https://github.com/marcel-licence/ML_SynthTools
I tried to made a little guide: https://youtu.be/c7TL8jcrnzs
Hope this helps

**Reply by mguibord** [reply]
Thank you!  I wasn't aware the board definitions were integrated into the ML_SynthTools library.

I have another error that fails during the linking process maybe you can shed some light as to why.

C:\Users\mguib\AppData\Local\Temp\arduino_build_766619\sketch\esp32_midi_sampler.ino.cpp.o:(.literal._Z18Screen_EffectsDrawv+0xc): undefined reference to `delayToMix'
C:\Users\mguib\AppData\Local\Temp\arduino_build_766619\sketch\esp32_midi_sampler.ino.cpp.o:(.literal._Z18Screen_EffectsDrawv+0x14): undefined reference to `delayInLvl'
C:\Users\mguib\AppData\Local\Temp\arduino_build_766619\sketch\esp32_midi_sampler.ino.cpp.o:(.literal._Z18Screen_EffectsDrawv+0x1c): undefined reference to `delayLen'
C:\Users\mguib\AppData\Local\Temp\arduino_build_766619\sketch\esp32_midi_sampler.ino.cpp.o:(.literal._Z18Screen_EffectsDrawv+0x28): undefined reference to `delayFeedback'
collect2.exe: error: ld returned 1 exit status

**Reply by marcel-licence** [reply]
I guess you tried to use DISPLAY_160x80_ENABLED. I tested the code the last time only without DISPLAY_160x80_ENABLED. 

A quick fix would be to replace:
```
extern float delayToMix;
extern float delayInLvl;
extern uint32_t delayLen;
extern float delayFeedback;
```
by
```
float delayToMix = 0;
float delayInLvl = 0;
uint32_t delayLen = 0;
float delayFeedback = 0;
```
in screens.ino.

The display implementation is highly experimental.
